src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt

```diff
@@ -45,27 +49,63 @@ namespace Microsoft.OData.ModelBuilder
     public static class VocabularyTermConfigurationExtensions
     {
         internal static IEdmExpression ToEdmExpression(this string text)
-            => string.IsNullOrEmpty(text) ? null: new EdmStringConstant(text);
+            => string.IsNullOrEmpty(text) ? null : new EdmStringConstant(text);
 <#
-        foreach (var term in model.SchemaElements.OfType<IEdmTerm>())
+        foreach(var model in models)
         {
-            var targets = term.AppliesTo.Split(' ').ToHashSet();
-            var sourceConfig = default(string);
-            if (targets.IsSupersetOf(new[]{ "Singleton", "EntitySet" }))
-            {
-                sourceConfig = "NavigationSourceConfiguration";
-            }
-            else if (targets.Contains("Singleton"))
-            {
-                sourceConfig = "SingletonConfiguration";
-            }
-            else if (targets.Contains("EntitySet"))
+            foreach (var term in model.SchemaElements.OfType<IEdmTerm>())
             {
-                sourceConfig = "EntitySetConfiguration";
-            }
+                var appliesTo = term.AppliesTo;
+                if(string.IsNullOrEmpty(appliesTo))
+                {
+                    if (term.Name == "Revisions")
+                    {
+                        appliesTo = "EntitySet Singleton Property";
```

Comment:
Do we need this as it seems like the default currently as @g2mula has pointed out can be applied to anything. I am guessing this can even be applied to other annotations with the current definition.